src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/llm_inputs.py

```diff
@@ -500,30 +495,30 @@ def _convert_generic_json_to_trtllm_format(
         return pa_json
 
     @classmethod
-    def _write_json_to_file(cls, json_in_pa_format: Dict):
-        try:
-            f = open(DEFAULT_INPUT_DATA_JSON, "w")
+    def _write_json_to_file(cls, json_in_pa_format: Dict) -> None:
+        with open(DEFAULT_INPUT_DATA_JSON, "w") as f:
             f.write(json.dumps(json_in_pa_format, indent=2))
-        finally:
-            f.close()
 
     @classmethod
     def _determine_json_feature_roles(
         cls, dataset_json: Dict
-    ) -> Tuple[List[str], List[str]]:
+    ) -> Tuple[List[str], List[str], List[str]]:
         SYSTEM_ROLE_LIST = ["system_prompt"]
         USER_ROLE_LIST = ["question", "article"]
         TEXT_INPUT_LIST = ["text_input"]
 
-        system_role_headers, user_role_headers, text_input_headers = [], [], []
+        system_role_headers: List[str] = []
+        user_role_headers: List[str] = []
+        text_input_headers: List[str] = []
+
         if "features" in dataset_json.keys():
             for index, feature in enumerate(dataset_json["features"]):
                 if feature in SYSTEM_ROLE_LIST:
                     system_role_headers.append(feature)
                 if feature in USER_ROLE_LIST:
                     user_role_headers.append(feature)
                 if feature in TEXT_INPUT_LIST:
-                    user_role_headers.append(feature)
+                    text_input_headers.append(feature)
```

Comment:
CC: @nv-braf. I made this change, which seems logical. If you see any issues, let me know. I'll run tests on the backends/endpoints.

Reply:
Reverting this fix, as it breaks OpenAI. We should fix this, as it's not doing what we expect it to be doing.

Ref: TMA-1847